modules/zarr/src/lib/storage.ts

```diff
@@ -0,0 +1,51 @@
+import { fetchFile } from '@loaders.gl/core';
+import { KeyError } from 'zarr';
+import type { AsyncStore } from 'zarr/types/storage/types';
+
+/**
+ * Preserves (double) slashes earlier in the path, so this works better
+ * for URLs. From https://stackoverflow.com/a/46427607/4178400
+ * @param args parts of a path or URL to join.
+ */
+function joinUrlParts(...args: string[]) {
+  return args
+    .map((part, i) => {
+      if (i === 0) return part.trim().replace(/[/]*$/g, '');
+      return part.trim().replace(/(^[/]*|[/]*$)/g, '');
+    })
+    .filter(x => x.length)
+    .join('/');
+}
+
+export class FetchFileStore implements AsyncStore<ArrayBuffer> {
+  constructor(public root: string) { }
+
+  async getItem(key: string): Promise<ArrayBuffer> {
+    const filepath = joinUrlParts(this.root, key);
+    const response = await fetchFile(filepath);
+    if (!response.ok) {
+      // Zarr requires a special exception to be thrown in case of missing chunks
+      throw new KeyError(key);
+    }
+    const value = await response.arrayBuffer();
+    return value;
+  }
+
+  async containsItem(key: string): Promise<boolean> {
+    const filepath = joinUrlParts(this.root, key);
+    const response = await fetchFile(filepath);
+    return response.ok;
+  }
+
+  keys(): Promise<string[]> {
+    return Promise.resolve([]);
+  }
+
+  setItem(): never {
+    throw new Error('setItem not implemented.');
```

Comment:
Note I believe @ibgreen tends to take out error strings for bundle size reasons.